src/DynamicData/List/SourceList.cs

```diff
@@ -160,10 +160,13 @@ public IObservable<IChangeSet<T>> Connect(Func<T, bool> predicate = null)
             {
                 lock (_locker)
                 {
-                    var initial = new ChangeSet<T>(new[] {new Change<T>(ListChangeReason.AddRange, _readerWriter.Items)});
-                    if (initial.TotalChanges > 0)
+                    if (_readerWriter.Items.Length > 0)
```

Comment:
Can short circuit if the current items are empty, rather than making a changeset and filling it first